src/components/TodoInput.js

```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+function TodoInput(props) {
+  return (
+    <div className="form-group form-styled">
```

Comment:
1. why don't you use `<form>` tag here with all the built-in validations like `required`?
2. Handle submit as `onSubmit` not as a a button click
3. Does parent component need to know about the "dirty" state of the form? What i'm saying is whenever you have some kind of validation on forms all that validation needs to be encapsulated in the form usually and only submit callback should be propagated to the parent. Here is an example

```
<div>
  <h1>Todo Items</h1>

  <ul>
    {todoItems.map(item => <li>{item.name}</li>)}
  </ul>

  <h1>Add a new item</h1>

  {/* inside the parent I dont care what kind of validation is done inside the Form usually */}
  {/* the only thing is care is how to add a value to my state on submit */}
  <AddItemForm onSubmit={newItem => /* code add a newItem to todoItems */ }>
</div>

```

Reply:
It's my serious mess,
I'm changing form to work with built-in validations, using this article as a guide: 
[Form_validation](https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation)